src/sys/wait.rs

```diff
@@ -215,24 +236,94 @@ fn decode(pid : Pid, status: i32) -> WaitStatus {
     }
 }
 
-pub fn waitpid<P: Into<Option<Pid>>>(pid: P, options: Option<WaitPidFlag>) -> Result<WaitStatus> {
-    use self::WaitStatus::*;
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash)]
+/// Designates whether the supplied `Pid` value is a process ID, process group ID,
+/// specifies any child of the current process's group ID, or any child of the current process.
+pub enum PidGroup {
+    ProcessID(u32),
```

Comment:
I'm not certain we're quite on the same page with understanding this issue based on your comment here: I'm trying to develop a solution where you never need to negate the `pid` as well. My understanding that when using `waitpid()` that you will already have a process ID, the question is what does it mean. That extra meaning is why there's the enum type, so even though `ProcessID(pid)` and `ProcessGroupId(pid)` have the same process id within them, they mean different things. And internally the values would be mapped as the `libc::waitpid` function expected. So the point is that the user shouldn't be doing the negation just as you suggested, the value used for `ProcessId` and `ProcessGroupId` is the same raw pid value.

What we do need to make sure is that the raw value is "correct". For example, the `pid` should be positive and non-zero for both of those types, so that should be checked. We can either embed this into the enum type (which I don't know a good implementation for) or embed it within `waitpid()`, which is the path I'd suggest.

Is that more clear? Because I think we agree we're just losing something in translation.